plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepositoryPlugin.java

```diff
@@ -100,6 +150,32 @@ public void reload(Settings settings) {
         if (clientsSettings.isEmpty()) {
             throw new SettingsException("If you want to use an azure repository, you need to define a client configuration.");
         }
-        azureStoreService.refreshAndClearCache(clientsSettings);
+        AzureStorageService storageService = azureStoreService.get();
+        assert storageService != null;
+        storageService.refreshSettings(clientsSettings);
+    }
+
+    @SuppressForbidden(reason = "needs to overcome the eager class loading problem")
+    private void setUpDefaultHttpClientProvider() {
```

Comment:
This is really unfortunate, the SDK loads a default http client provider during the class construction and since the dependency jars are loaded in lexicographical order, the netty http client is not loaded at that time. This is a bug on the blob batch client, that instead of reusing the existing http client, it creates a new one. I've opened a PR and it's already approved, so this is just a temporal workaround.